pkg/providers/ovirt/mapper/mapper.go

```diff
@@ -206,5 +279,168 @@ func (o *OvirtMapper) mapCPU() *kubevirtv1.CPU {
+	// Guest memory
+	ovirtVMMemory, _ := o.vm.Memory()
+	guestMemory, _ := resource.ParseQuantity(strconv.FormatInt(ovirtVMMemory, 10))
+	memory := &kubevirtv1.Memory{}
+	memory.Guest = &guestMemory
+
+	// HugePages
+	hp := o.mapHugePages()
+	if hp != nil {
+		memory.Hugepages = hp
+	}
+
+	return memory
+}
+
+func (o *OvirtMapper) mapHugePages() *kubevirtv1.Hugepages {
+	var hugePageSize *string
+	if customProperties, available := o.vm.CustomProperties(); available {
+		for _, cp := range customProperties.Slice() {
+			if cpName, _ := cp.Name(); cpName == customPropertyHugePages {
+				cmValue, _ := cp.Value()
+				hugePageSize = &cmValue
+				break
+			}
+		}
+	}
+	if hugePageSize != nil {
+		return &kubevirtv1.Hugepages{
+			// HugePage size in oVirt custom_property is in MiB, so we add Mi suffix here.
+			PageSize: fmt.Sprintf("%sMi", *hugePageSize),
+		}
+	}
+
+	return nil
+}
+
+func (o *OvirtMapper) mapResourceRequirements() kubevirtv1.ResourceRequirements {
+	reqs := kubevirtv1.ResourceRequirements{}
+	memoryPolicy, _ := o.vm.MemoryPolicy()
+	maxMemory, _ := memoryPolicy.Max()
+	maxMemoryQuantity, _ := resource.ParseQuantity(strconv.FormatInt(maxMemory, 10))
+	reqs.Limits = map[corev1.ResourceName]resource.Quantity{
+		corev1.ResourceMemory: maxMemoryQuantity,
+	}
+
+	return reqs
+}
+
+// Graphical console is attached only in case the VM is not in headless mode
+func (o *OvirtMapper) mapGraphicalConsoles() *kubevirtv1.Devices {
+	// GraphicsConsole
+	vncEnabled := true
+	if gc, _ := o.vm.GraphicsConsoles(); len(gc.Slice()) == 0 {
+		vncEnabled = false
+	}
+	devices := &kubevirtv1.Devices{}
+	devices.AutoattachGraphicsDevice = &vncEnabled
+
+	// SerialConsole
+	if console, ok := o.vm.Console(); ok {
+		consoleEnabled, _ := console.Enabled()
+		devices.AutoattachSerialConsole = &consoleEnabled
+	}
+
+	return devices
+}
+
+func (o *OvirtMapper) mapLabels() map[string]string {
+	labels := map[string]string{}
+
+	// Origin
+	labels[LabelOrigin], _ = o.vm.Origin()
+
+	// Instance type
+	if instanceType, ok := o.vm.InstanceType(); ok {
+		labels[LabelInstanceType], _ = instanceType.Name()
+	}
+
+	// Tags
+	if tags, ok := o.vm.Tags(); ok {
+		var tagList []string
+		for _, tag := range tags.Slice() {
+			tagName, _ := tag.Name()
+			tagList = append(tagList, tagName)
+		}
+		labels[LabelTag] = strings.Join(tagList, ",")
+	}
+
+	// Fqdn
+	if fqdn, ok := o.vm.Fqdn(); ok {
+		labels[LabelFQDN] = fqdn
+	}
+
+	return labels
+}
+
+func (o *OvirtMapper) mapAnnotations() map[string]string {
+	annotations := map[string]string{}
+
+	// Comment
+	if comment, ok := o.vm.Comment(); ok {
+		annotations[AnnotationComment] = comment
+	}
+
+	// SSO
+	if sso, ok := o.vm.Sso(); ok {
+		if methods, _ := sso.Methods(); len(methods.Slice()) > 0 {
+			annotations[AnnotationSso] = string(ovirtsdk.SSOMETHOD_GUEST_AGENT)
+		} else {
+			annotations[AnnotationSso] = "disabled"
+		}
+	}
+
+	// Errata
+	var errataValue []string
+	if errata, ok := o.vm.KatelloErrata(); ok {
```

Comment:
I don't think we need to map katello errata.
If so, the vm will have to be registered to Satellite/Katello and use it for viewing the content/errata available for the VM.
Also, If we set that as part of the annotation, I'm not aware of a process that will make sure to remove/update when errata is applied for the VM.

Reply:
Well, it was defined in the document so I did that. I think it's not that important as well, so I've removed it. We can add it later if needed.